AEPAssurance/Source/AssuranceEventChunker.swift

```diff
@@ -102,4 +110,43 @@ struct AssuranceEventChunker {
         }
         return chunkedEvents
     }
+    
+    ///
+    /// Stitches chunked events together into one `AssuranceEvent` where the stitched events chunkData is now the new event's payload
+    ///
+    /// - Parameter chunkedEvents: An array of chunked AssuranceEvents
+    /// - Returns: An `AssuranceEvent` which has the stitched data as the payload
+    func stitch(_ chunkedEvents: [AssuranceEvent]) -> AssuranceEvent? {
+        var stitchedString = ""
+        // Stitch chunked payload data together
+        for event in chunkedEvents {
+            // Extract the payload string and unescape it. Currently escaped by the services
+            guard let payloadString = event.payload?[AssuranceConstants.AssuranceEvent.PayloadKey.CHUNK_DATA]?.stringValue else {
+                Log.error(label: AssuranceConstants.LOG_TAG, "Error while attempting to stitch chunked event: Chunk payload was not in proper string format.")
+                return nil
+            }
+            
+            stitchedString += payloadString
+        }
+        guard let stitchedStringData = stitchedString.data(using: .utf8) else {
+            Log.error(label: AssuranceConstants.LOG_TAG, "Error while attempting to create data from stitched string")
+            return nil
+        }
+        
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .millisecondsSince1970
+        var decodedPayload: [String: AnyCodable]? = nil
+        do {
+            decodedPayload = try decoder.decode([String:AnyCodable].self, from: stitchedStringData)
+        } catch {
+            Log.error(label: AssuranceConstants.LOG_TAG, "Error while attempting to decode stitched JSON data: \(error)")
+            return nil
+        }
+        let referenceEvent = chunkedEvents[0]
```

Comment:
Is it guaranteed that the Array will have atleast one item? Can you add a precondition?

Reply:
It technically is guaranteed, but doesn't hurt to put a precondition. Will do.